crates/bevy_text/src/text2d.rs

```diff
@@ -81,7 +83,15 @@ pub fn extract_text2d_sprite(
         )>,
     >,
 ) {
-    let scale_factor = windows.scale_factor(WindowId::primary()) as f32;
+    let resolution = windows
+        .get(
+            primary_window
+                .as_ref()
+                .expect("Primary window should exist")
+                .window,
+        )
+        .expect("Primary windows should have a valid WindowResolution component");
+    let scale_factor = resolution.scale_factor() as f32;
```

Comment:
Here we incorrectly assume just one window, but changing this is out of scope for this PR. 
See #5621

This should be solved in follow up prs.